networking/src/NTP/Client.hs

```diff
@@ -148,7 +149,9 @@ startSend addrs cli = do
     let poll    = ntpPollDelay (ncSettings cli)
     logDebug "Sending requests"
     liftIO . atomically . modifyTVarS (ncState cli) $ identity .= Just []
-    () <$ threadDelay timeout `race` forConcurrently addrs (flip doSend cli)
+    let sendRequests = forConcurrently addrs (flip doSend cli)
+    withAsync sendRequests $ \_ ->
+        () <$ threadDelay timeout
```

Comment:
I do not understand this change. Probably I am missing something. Suppose that `timeout` is 5 sec and `sendRequests` takes 1 sec. In previous implementation we would finish in 1 sec, but this implementation will always wait for 5 sec. Is it what we want?

Reply:
`sendRequests` actually does only "send" part, receiving occurs in another thread. And we should give some time for servers to respond.

Reply:
Uhm, I just tried the endpoint out and I don't think the current design is a robust one, because these endpoints are called fairly frequently and we cannot afford every call to take _at least_ 5 seconds.

Cannot we have a sort of `race` combinator like (literally the first thing which comes to my mind):

```
race (readMVar syncMVarForWorkers) (threadDelay timeout) 
``` 

So that as soon as the NTP workers finishes, we return?

Also, ancillary question: what happens if none of the workers respond? What do we return to the user?

Reply:
That's what I did in separate PR (it's not part of the fix):
https://github.com/input-output-hk/cardano-sl/pull/2449

If none of workers respond, then callback is not called and `MVar` is not filled till the next round of NTP requests

Reply:
Yes, this makes no sense.

The original version did this: A first to finish sync (with cancellation) between a timeout and (a last to finish sync between all sends).

The proposed version does the same last to finish sync between all sends, but now the sync with the timeout is asymmetric: it always waits for the full timeout time.

This makes no sense at all. If the problem is that we want to "wait for some time to get responses" then we should do that, which means doing it as part of waiting for the responses, not part of sending the requests.

So, what policy do we want? Do we want to wait for just one, or all? Of course we also need a timeout, but before the timeout should we wait for the first or would we prefer to wait (up to the timeout) for all of them, or some number of them?